src/rlookup.c

```diff
@@ -332,24 +333,21 @@ static int jsonIterToValue(RedisModuleCtx *ctx, JSONResultsIterator iter, unsign
     
     // Second, get the first JSON value
     RedisJSON json = japi->next(iter);
-    if (!json) {
-      goto cleanup;
+    if (json) {
+      RSValue *val = jsonValToValue(ctx, json);
+      RSValue *otherval = RS_StealRedisStringVal(serialized);
+      *rsv = RS_DuoVal(val, otherval);
+      res = REDISMODULE_OK;
+    } else if (serialized) {
+      RedisModule_FreeString(ctx, serialized);
     }
-    RSValue *val = jsonValToValue(ctx, json);
-    RSValue *otherval = RS_StealRedisStringVal(serialized);
-    *rsv = RS_DuoVal(val, otherval);
-    res = REDISMODULE_OK;
   }
   
 done:
-  japi->freeIter(iter);
-  return res;
-
-cleanup:
-  if (serialized) {
-    RedisModule_FreeString(ctx, serialized);
+  if (freeIter) {
```

Comment:
Why not free it from outside (when needed) and avoid the extra argument to the function?